### PR TITLE
[zh-cn]: init the translation of `FileSystemDirectoryHandle`

### PR DESCRIPTION
**Pull request by BSPR0002**
### Motivation
Initial zh-CN translation of `FileSystemDirectoryHandle`.

### Related issues and pull requests
Depends on [mdn/content pr#29012](https://github.com/mdn/content/pull/29012)

### CODE REVIEW

File: files/zh-cn/web/api/filesystemdirectoryhandle/index.md [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+- {{domxref('FileSystemDirectoryHandle.getDirectoryHandle()')}}
+  - : 返回一个 {{jsxref('Promise')}}，兑现一个调用此方法的目录句柄内指定名称的子目录的 {{domxref('FileSystemDirectoryHandle')}}。
+- {{domxref('FileSystemDirectoryHandle.getFileHandle()')}}
+  - : 返回一个 {{jsxref('Promise')}}，兑现一个调用此方法的目录句柄内指定名称的文件的 {{domxref('FileSystemFileHandle')}}。
+- {{domxref('FileSystemDirectoryHandle.removeEntry()')}}
+  - : 如果目录句柄包含一个名为指定名称的文件或目录，则尝试异步将其移除。
+- {{domxref('FileSystemDirectoryHandle.resolve()')}}
+  - : 返回一个 {{jsxref('Promise')}} 对象，兑现一个包含从父目录前往指定子条目中间的目录的名称的{{jsxref('Array', '数组', '', 'nocode')}}。数组的最后一项是子条目的名称。
+
+[异步迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#the_async_iterator_and_async_iterable_protocols)方法:
```
Comment (BSPR0002) [inline files/zh-cn/web/api/filesystemdirectoryhandle/index.md:33]:
```suggestion
[异步迭代器](/zh-CN/docs/Web/JavaScript/Reference/Iteration_protocols#the_async_iterator_and_async_iterable_protocols)方法：
```

**Comment by yin1999** [comment]
Hey @BSPR0002. Could you update this PR as the upstream changes are merged :)

File: files/zh-cn/web/api/filesystemdirectoryhandle/getdirectoryhandle/index.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+- `options` {{optional_inline}}
+
+  - : 包含要获得的子目录的选项的可选对象，选项如下：
+
+    - `create`
+      - : 布尔值，默认为 `false`。当设为 `true` 时，如果没有找到对应的目录，将会创建一个指定名称的目录并将其返回。
+
+### 返回值
+
+一个 {{jsxref('Promise')}} 对象，兑现一个 {{domxref('FileSystemDirectoryHandle')}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/getdirectoryhandle/index.md:30]: fulfillment happens "in the future"

```suggestion
一个 {{jsxref('Promise')}} 对象，会兑现一个 {{domxref('FileSystemDirectoryHandle')}}。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/getdirectoryhandle/index.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+### 异常
+
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : 如果 {{domxref('PermissionStatus')}} 不为 'granted' 则抛出此异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/getdirectoryhandle/index.md:35]:
```suggestion
  - : 如果 {{domxref('PermissionStatus')}} 不为“granted”则抛出此异常。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/getfilehandle/index.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      - : 布尔值，默认为 `false`。当设为 `true` 时，如果没有找到对应的文件，将会创建一个指定名称的文件并将其返回。
+
+### 返回值
+
+一个 {{jsxref('Promise')}} 对象，兑现一个 {{domxref('FileSystemFileHandle')}}。
+
+### 异常
+
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : 如果 {{domxref('PermissionStatus')}} 不为 'granted' 则抛出此异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/getfilehandle/index.md:35]:
```suggestion
  - : 如果 {{domxref('PermissionStatus')}} 不为“granted”则抛出此异常。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/getfilehandle/index.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+- `options` {{optional_inline}}
+
+  - : 包含以下属性的对象：
+
+    - `create`
+      - : 布尔值，默认为 `false`。当设为 `true` 时，如果没有找到对应的文件，将会创建一个指定名称的文件并将其返回。
+
+### 返回值
+
+一个 {{jsxref('Promise')}} 对象，兑现一个 {{domxref('FileSystemFileHandle')}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/getfilehandle/index.md:30]:
```suggestion
一个 {{jsxref('Promise')}} 对象，会兑现一个 {{domxref('FileSystemFileHandle')}}。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/index.md [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+## 实例方法
+
+_从父接口 {{DOMxRef("FileSystemHandle")}} 继承方法。_
+
+常规方法：
+
+- {{domxref('FileSystemDirectoryHandle.getDirectoryHandle()')}}
+  - : 返回一个 {{jsxref('Promise')}}，兑现一个调用此方法的目录句柄内指定名称的子目录的 {{domxref('FileSystemDirectoryHandle')}}。
+- {{domxref('FileSystemDirectoryHandle.getFileHandle()')}}
+  - : 返回一个 {{jsxref('Promise')}}，兑现一个调用此方法的目录句柄内指定名称的文件的 {{domxref('FileSystemFileHandle')}}。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/index.md:27]: Could we remove the self-referenced link (also in en-US)?

```suggestion
  - : 返回一个 {{jsxref('Promise')}}，会兑现一个调用此方法的目录句柄内指定名称的子目录的 {{domxref('FileSystemDirectoryHandle')}}。
- {{domxref('FileSystemDirectoryHandle.getFileHandle()')}}
  - : 返回一个 {{jsxref('Promise')}}，会兑现一个调用此方法的目录句柄内指定名称的文件的 `FileSystemFileHandle`。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/index.md [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+- {{domxref('FileSystemDirectoryHandle.removeEntry()')}}
+  - : 如果目录句柄包含一个名为指定名称的文件或目录，则尝试异步将其移除。
+- {{domxref('FileSystemDirectoryHandle.resolve()')}}
+  - : 返回一个 {{jsxref('Promise')}} 对象，兑现一个包含从父目录前往指定子条目中间的目录的名称的{{jsxref('Array', '数组', '', 'nocode')}}。数组的最后一项是子条目的名称。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/index.md:31]:
```suggestion
  - : 返回一个 {{jsxref('Promise')}} 对象，会兑现一个包含从父目录前往指定子条目中间的目录的名称的{{jsxref('Array', '数组', '', 'nocode')}}。数组的最后一项是子条目的名称。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/removeentry/index.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+### 返回值
+
+一个 {{jsxref('Promise')}} 对象，兑现 `undefined`。
+
+### 异常
+
+- {{jsxref("TypeError")}}
+  - : 如果 `name` 不是一个合法的字符串或者包含不允许出现在文件系统中的字符时，抛出此异常。
+- `NotAllowedError` {{domxref("DOMException")}}
+  - : 如果 {{domxref('PermissionStatus')}} 不为 'granted' 则抛出此异常。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/removeentry/index.md:37]:
```suggestion
  - : 如果 {{domxref('PermissionStatus')}} 不为“granted”则抛出此异常。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/resolve/index.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
+---
+title: FileSystemDirectoryHandle：resolve() 方法
+slug: Web/API/FileSystemDirectoryHandle/resolve
+---
+
+{{securecontext_header}}{{APIRef("File System API")}}
+
+{{domxref("FileSystemDirectoryHandle")}} 接口的 **`resolve()`** 方法返回一个 {{jsxref('Promise')}} 对象，兑现一个包含从父目录前往指定子条目中间的目录的名称的{{jsxref('Array', '数组', '', 'nocode')}}。数组的最后一项是子条目的名称。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/resolve/index.md:8]:
```suggestion
{{domxref("FileSystemDirectoryHandle")}} 接口的 **`resolve()`** 方法返回一个 {{jsxref('Promise')}} 对象，会兑现一个包含从父目录前往指定子条目中间的目录的名称的{{jsxref('Array', '数组', '', 'nocode')}}。数组的最后一项是子条目的名称。
```

File: files/zh-cn/web/api/filesystemdirectoryhandle/resolve/index.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+```
+
+### 参数
+
+- `possibleDescendant`
+  - : 要返回其相对路径的 {{domxref('FileSystemHandle')}}。
+
+### 返回值
+
+一个 {{jsxref('Promise')}} 对象，兑现一个包含字符串的{{jsxref('Array', '数组', '', 'nocode')}}，或者当参数 `possibleDescendant` 不是此 {{domxref('FileSystemDirectoryHandle')}} 的后代时，兑现 `null`。
```
Comment (yin1999) [inline files/zh-cn/web/api/filesystemdirectoryhandle/resolve/index.md:23]:
```suggestion
一个 {{jsxref('Promise')}} 对象，会兑现一个包含字符串的{{jsxref('Array', '数组', '', 'nocode')}}，或者当参数 `possibleDescendant` 不是此 {{domxref('FileSystemDirectoryHandle')}} 的后代时，兑现 `null`。
```

**Review by yin1999 — COMMENTED** [review]
Great. Just some comments

**Review by yin1999 — APPROVED** [review]
Thank you @BSPR0002

**Pull request merged**